nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,840 @@
+                 low_pass=None, high_pass=None, t_r=None,
+                 mask_strategy='background', is_classification=True,
+                 memory=None, memory_level=0, n_jobs=1, verbose=0):
+        self.estimator = estimator
+        self.mask = mask
+        self.cv = cv
+        self.param_grid = param_grid
+        self.screening_percentile = screening_percentile
+        self.scoring = scoring
+        self.is_classification = is_classification
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+
+    def fit(self, X, y, groups=None):
+        """Fit the decoder (learner).
+
+        Parameters
+        ----------
+        X: list of Niimg-like objects
+            See http://nilearn.github.io/manipulating_images/input_output.html
+            Data on which model is to be fitted. If this is a list,
+            the affine is considered the same for all.
+
+        y: numpy.ndarray of shape=(n_samples) or list of length n_samples
+            The dependent variable (age, sex, IQ, yes/no, etc.).
+            Target variable to predict. Must have exactly as many elements as
+            3D images in niimg.
+
+        groups: None or bool, optional
+            Must be specify in multi-label classification settings.
+            Group labels for the samples used while splitting the dataset into
+            train/test set. This ‘groups’ parameter must always be specified to
+            calculate the number of splits.
+
+        Attributes
+        ----------
+        `masker_`: instance of NiftiMasker or MultiNiftiMasker
+            The NiftiMasker used to mask the data.
+
+        `mask_img_`: Nifti1Image
+            Mask computed by the masker object.
+
+        `classes_`: numpy.ndarray
+            Classes to predict. For classification only.
+
+        `screening_percentile_`: float
+            Screening percentile corrected according to volume of mask,
+            relative to the volume of standard brain.
+
+        `coef_`: numpy.ndarray, shape=(n_classes, n_features)
+            Contains the mean of the models weight vector across
+            fold for each class.
+
+        `coef_img_`: dict of Nifti1Image
+            Dictionary containing `coef_` with class names as keys,
+            and `coef_` transformed in Nifti1Images as values. In the case
+            of a regression, it contains a single Nifti1Image at the key 'beta'.
+
+        `intercept_`: narray, shape (nclasses -1,)
+            Intercept (a.k.a. bias) added to the decision function.
+            It is available only when parameter intercept is set to True.
```

Comment:
there is no such parameter so it is always available. why nclassses - 1?

Reply:
>  It is available only when parameter intercept is set to True. 
I guess the purpose for this line from original author is that when user using customized scikit-learn estimator they might get the one that sets `intercept=False`.

Reply:
Ok I will delete this line anyway because it does not make sense to do regression/classification without intercept.